sagemaker-neo-notebooks/edge/cpp-integration/tutorial.cc

```diff
@@ -0,0 +1,622 @@
+    Aws::String aws_policy_arn = policies[i];
+    Aws::IAM::Model::AttachRolePolicyRequest attach_policy_request;
+    attach_policy_request.SetPolicyArn(aws_policy_arn);
+    attach_policy_request.SetRoleName(aws_role_name);
+    auto attach_policy_outcome =
+        iam_client.AttachRolePolicy(attach_policy_request);
+    if (!attach_policy_outcome.IsSuccess()) {
+      auto error = attach_policy_outcome.GetError();
+      const std::string error_str = getErrorMessage(error);
+      throw std::runtime_error("AttachIamPolicy error : " + error_str);
+    }
+  }
+}
+
+void CreateIamStep() {
+  if (!checkIamRole(ROLE_NAME)) {
+    createIamRole(ROLE_NAME);
+    attachIamPolicy(ROLE_NAME);
+  }
+}
+
+string getJobName() {
+  chrono::milliseconds ms = chrono::duration_cast<chrono::milliseconds>(
+      chrono::system_clock::now().time_since_epoch());
+
+  std::stringstream ss;
+  ss << "pi-demo-" << std::to_string(ms.count());
+  return ss.str();
+}
+
+Aws::SageMaker::Model::CompilationJobStatus poll_job_status(string job_name) {
+  Aws::SageMaker::SageMakerClient sm_client = getSageMakerClient();
+  Aws::SageMaker::Model::DescribeCompilationJobRequest describe_job_request;
+
+  Aws::String aws_job_name(job_name.c_str(), job_name.size());
+  describe_job_request.SetCompilationJobName(aws_job_name);
+  auto describe_job_resp =
+      sm_client.DescribeCompilationJob(describe_job_request);
+
+  if (!describe_job_resp.IsSuccess()) {
+    auto error = describe_job_resp.GetError();
+    const std::string error_str = getErrorMessage(error);
+    throw std::runtime_error("DescribeCompilationJob error: " + error_str);
+  }
+
+  auto result = describe_job_resp.GetResult();
+  return result.GetCompilationJobStatus();
+}
+
+void getIamRole(string role_name, Aws::IAM::Model::Role &role) {
+  Aws::IAM::IAMClient iam_client = getIamClient();
+  Aws::String aws_iam_role(role_name.c_str(), role_name.size());
+  Aws::IAM::Model::GetRoleRequest get_role_request;
+  get_role_request.SetRoleName(aws_iam_role);
+
+  auto get_role_response = iam_client.GetRole(get_role_request);
+  if (get_role_response.IsSuccess()) {
+    role = get_role_response.GetResult().GetRole();
+  } else {
+    auto error = get_role_response.GetError();
+    auto errorType = error.GetErrorType();
+    if (errorType == Aws::IAM::IAMErrors::NO_SUCH_ENTITY) {
+      throw std::runtime_error("Role doesn't exist and will create one");
+    } else {
+      const std::string error_str = getErrorMessage(error);
+      throw std::runtime_error("GetIamRole error: " + error_str);
+    }
+  }
+}
+
+void CompileNeoModel(string &bucket_name, string &model_name, string &target) {
+  // set input parameters
+  const string input_s3 = "s3://" + bucket_name + "/" + model_name;
+
+  const Aws::SageMaker::Model::Framework framework =
+      Aws::SageMaker::Model::Framework::MXNET;
+  const Aws::String aws_s3_uri(input_s3.c_str(), input_s3.size());
+  const Aws::String data_shape = "{\"data\":[1,3,224,224]}";
```

Comment:
single quotes (not pure json) are also supported by Neo Service `"{'data':[1,3,224,224]}"`